PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/PrebidServerAdapter.java

```diff
@@ -560,14 +559,117 @@ private JSONArray getImp() throws NoContextException {
                     JSONArray format = new JSONArray();
                     for (AdSize size : requestParams.getAdSizes()) {
                         format.put(new JSONObject().put("w", size.getWidth()).put("h", size.getHeight()));
                     }
                     banner.put("format", format);
                     imp.put("banner", banner);
+                } else if (requestParams.getAdType().equals(AdType.NATIVE)) {
+                    // add native request
+                    JSONObject nativeObj = new JSONObject();
+                    JSONObject request = new JSONObject();
+                    JSONArray assets = new JSONArray();
+                    HashMap<String, Object> requestConfig = requestParams.getNativeRequestAssets();
+                    if (requestConfig.get(NativeAdUnit.CONTEXT) != null) {
+                        request.put(NativeAdUnit.CONTEXT, requestConfig.get(NativeAdUnit.CONTEXT));
+                    }
+                    if (requestConfig.get(NativeAdUnit.CONTEXT_SUB_TYPE) != null) {
+                        request.put(NativeAdUnit.CONTEXT_SUB_TYPE, requestConfig.get(NativeAdUnit.CONTEXT_SUB_TYPE));
+                    }
+                    if (requestConfig.get(NativeAdUnit.PLACEMENT_TYPE) != null) {
+                        request.put(NativeAdUnit.PLACEMENT_TYPE, requestConfig.get(NativeAdUnit.PLACEMENT_TYPE));
+                    }
+                    if (requestConfig.get(NativeAdUnit.PLACEMENT_COUNT) != null) {
+                        request.put(NativeAdUnit.PLACEMENT_COUNT, requestConfig.get(NativeAdUnit.PLACEMENT_COUNT));
+                    }
+                    if (requestConfig.get(NativeAdUnit.SEQ) != null) {
+                        request.put(NativeAdUnit.SEQ, requestConfig.get(NativeAdUnit.SEQ));
+                    }
+                    if (requestConfig.get(NativeAdUnit.A_URL_SUPPORT) != null) {
+                        request.put(NativeAdUnit.A_URL_SUPPORT, requestConfig.get(NativeAdUnit.A_URL_SUPPORT));
+                    }
+                    if (requestConfig.get(NativeAdUnit.D_URL_SUPPORT) != null) {
+                        request.put(NativeAdUnit.D_URL_SUPPORT, requestConfig.get(NativeAdUnit.D_URL_SUPPORT));
+                    }
+                    JSONArray trackers = (JSONArray) requestConfig.get(NativeAdUnit.EVENT_TRACKERS);
+                    if (trackers != null && trackers.length() > 0) {
+                        request.put(NativeAdUnit.EVENT_TRACKERS, requestConfig.get(NativeAdUnit.EVENT_TRACKERS));
+                    }
+                    if (requestConfig.get(NativeAdUnit.PRIVACY) != null) {
+                        request.put(NativeAdUnit.PRIVACY, requestConfig.get(NativeAdUnit.PRIVACY));
+                    }
+                    if (requestConfig.get(NativeAdUnit.EXT) != null) {
+                        request.put(NativeAdUnit.EXT, requestConfig.get(NativeAdUnit.EXT));
+                    }
```

Comment:
It is safe to remove `if` (null check) before putting something into JSONObject.

Reply:
This is semi-final of the PR, @ppuviarasu is working on iOS version, I'll wait on feedbacks from her implementation.
I'll update based on your feedback too.

Reply:
If it's null, the generated JSON object will just have a empty field right?

Reply:
doc says:
>If the value is null, any existing mapping for name is removed.